src/components/NavBar.js

```diff
@@ -1,33 +1,45 @@
-import React, { Component } from 'react';
-import { NavLink } from 'react-router-dom';
-import { Container, Menu, Input } from 'semantic-ui-react';
-import '../styling/Navbar.css';
-import AlertMessage from './AlertMessage';
-import { connect } from 'react-redux';
+import React, { Component } from "react";
+import { NavLink } from "react-router-dom";
+import { Container, Menu, Input } from "semantic-ui-react";
+import "../styling/Navbar.css";
+import AlertMessage from "./AlertMessage";
+import { connect } from "react-redux";
+import UpdateLanguage from './UpdateLanguage';
+import i18n from "../i18n";
 
 class NavBar extends Component {
-  state = { activeItem: 'latest news' }
+  constructor(props) {
+    super(props)
+    this.ReRender = this.ReRender.bind(this);
+  };
+  handleItemClick = (e, { name }) => this.setState({ activeItem: name });
 
-  handleItemClick = (e, { name }) => this.setState({ activeItem: name })
+  ReRender() {
+    this.forceUpdate();
+  }
 
   render() {
     let createArticleButton;
     let subscribeButton;
     let flashMessage;
     let loginActions;
 
-    const { activeItem } = this.state
+    const { t } = this.props;
```

Comment:
I assume that `t` stands for translation, yet you use `i18n.t` in your jsx.

Reply:
Yes, {t} is unnecessary as we call t directly from i18n.t, and has been removed.